infra/cifuzz/run_fuzzers.py

```diff
@@ -27,61 +27,167 @@
 import utils
 
 
+class BaseFuzzTargetRunner:
+  """Base class for fuzzer runners."""
+
+  def __init__(self, config):
+    self.config = config
+    # Set by the initialize method.
+    self.out_dir = None
+    self.fuzz_target_paths = None
+    self.artifacts_dir = None
+
+  def initialize(self):
+    """Initialization method. Must be called before calling run_fuzz_targets.
+    Returns True on success."""
+    # Use a seperate initialization function so we can return False on failure
+    # instead of exceptioning like we need to do if this were done in the
+    # __init__ method.
+
+    logging.info('Using %s sanitizer.', self.config.sanitizer)
+
+    # TODO(metzman) Add a check to ensure we aren't over time limit.
+    if not self.config.fuzz_seconds or self.config.fuzz_seconds < 1:
+      logging.error(
+          'Fuzz_seconds argument must be greater than 1, but was: %s.',
+          self.config.fuzz_seconds)
+      return False
+
+    self.out_dir = os.path.join(self.config.workspace, 'out')
+    if not os.path.exists(self.out_dir):
+      logging.error('Out directory: %s does not exist.', self.out_dir)
+      return False
+
+    self.artifacts_dir = os.path.join(self.out_dir, 'artifacts')
+    if not os.path.exists(self.artifacts_dir):
+      os.mkdir(self.artifacts_dir)
+    elif (not os.path.isdir(self.artifacts_dir) or
+          os.listdir(self.artifacts_dir)):
+      logging.error('Artifacts path: %s exists and is not an empty directory.',
+                    self.artifacts_dir)
+      return False
+
+    self.fuzz_target_paths = utils.get_fuzz_targets(self.out_dir)
+    logging.info('Fuzz targets: %s', self.fuzz_target_paths)
+    if not self.fuzz_target_paths:
+      logging.error('No fuzz targets were found in out directory: %s.',
+                    self.out_dir)
+      return False
+
+    return True
+
+  def run_fuzz_target(self, fuzz_target_obj):  # pylint: disable=no-self-use
+    """Fuzzes with |fuzz_target_obj| and returns the result."""
+    # TODO(metzman): Make children implement this so that the batch runner can
+    # do things differently.
+    return fuzz_target_obj.fuzz()
```

Comment:
Do we need to print some message that fuzzer is not found or log_error ?

Reply:
Nope, not here. The purpose of this function will become clearer later on when BatchFuzzerRunner starts overriding it.